chromatic_fitting.py

```diff
@@ -78,127 +86,277 @@ def summarize_parameters(self):
+                    k = f"wavelength_{j+i}"
+                    # mu = Deterministic(f'{k}_mu',self.every_light_curve[k])
+                    pm.Normal(
+                        f"{k}_data",
+                        mu=self.every_light_curve[k],
+                        sd=data.uncertainty[i, :],
+                        observed=data.flux[i, :],
+                    )
 
     def sample_prior(self, ndraws=3):
         """
         Draw samples from the prior distribution.
         :parameter n
         Number of priors to sample
         """
-        with self.model:
-            return sample_prior_predictive(ndraws)
+        try:
+            with self.pymc3_model:
+                return sample_prior_predictive(ndraws)
+        except:
+            priors = []
+            for mod in self.pymc3_model:
+                with mod:
+                    priors.append(sample_prior_predictive(ndraws))
+            return priors
 
-    def sample_posterior(self, trace, ndraws=3):
+    def sample_posterior(self, ndraws=3):
         """
         Draw samples from the posterior distribution.
         :parameter n
         Number of posteriors to sample
         """
-        with self.model:
-            return sample_posterior_predictive(trace, ndraws)
+        if not hasattr(self, 'trace'):
+            print("Sampling has not been run yet! Running now with defaults...")
+            self.sample()
+
+        try:
+            with self.pymc3_model:
+                return sample_posterior_predictive(self.trace, ndraws)
+        except:
+            posteriors = []
+            for mod, trace in zip(self.pymc3_model, self.trace):
+                with mod:
+                    posteriors.append(sample_posterior_predictive(trace, ndraws))
+            return posteriors
+
+
+    def optimize(self, **kw):
+        """
+        Wrapper for PyMC3_ext sample
+        """
+        with self.pymc3_model:
+            return optimize(**kw)
 
     def sample(self, **kw):
         """
         Wrapper for PyMC3_ext sample
         """
-        with self.model:
-            self.trace = sample(**kw)
+        if self.optimization == "separate":
+            self.trace = []
+            for mod in self.pymc3_model:
+                with mod:
+                    self.trace.append(sample(**kw))
+        else:
+            with self.pymc3_model:
+                self.trace = sample(**kw)
 
     def summarize(self, **kw):
         """
         Wrapper for arviz summary
         """
-        with self.model:
-            self.summary = summary(self.trace, **kw)
+        if not hasattr(self, 'trace'):
+            print("Sampling has not been run yet! Running now with defaults...")
+            self.sample()
+
+        if self.optimization == "separate":
+            self.summary = []
+            for mod, trace in zip(self.pymc3_model, self.trace):
+                with mod:
+                    self.summary.append(summary(trace, **kw))
```

Comment:
Oh, cool! Does this give you a result that you can compare directly to a simultaneous fit?

Reply:
If you use `extract_from_posteriors()` then I think yes!